calico_node/tests/st/policy/test_profile.py

```diff
@@ -164,6 +168,52 @@ def test_rules_source_ip_nets(self):
                                  pass_list=self.n1_workloads + self.n2_workloads)
     test_rules_source_ip_nets.batchnumber = 4
 
+    def test_rules_source_ip_sets(self):
+        """Test global network sets end-to-end"""
+        # Add a rule to each profile that allows traffic from all the workloads in the *other*
+        # network by means of a network set (which would normally be blocked).
+        prof_n1, prof_n2 = self._get_profiles(self.new_profiles)
+        n1_ips = [str(workload.ip) + "/32" for workload in self.n1_workloads]
+        n2_ips = [str(workload.ip) + "/32" for workload in self.n2_workloads]
+
+        netset = {
+            'apiVersion': 'projectcalico.org/v3',
+            'kind': 'GlobalNetworkSet',
+            'metadata': {
+                'name': "netset-1",
+                'labels': {"group": "n1"},
+            },
+            'spec': {
+                'nets': n1_ips,
+            }
+        }
+        self.host1.writefile("netset.yaml", yaml.dump(netset, default_flow_style=False))
+        self.host1.calicoctl("create -f netset.yaml")
+        netset = {
+            'apiVersion': 'projectcalico.org/v3',
+            'kind': 'GlobalNetworkSet',
+            'metadata': {
+                'name': "netset-2",
+                'labels': {"group": "n2"},
+            },
+            'spec': {
+                'nets': n2_ips,
+            }
+        }
+        self.host1.writefile("netset.yaml", yaml.dump(netset, default_flow_style=False))
+        self.host1.calicoctl("create -f netset.yaml")
+
+        rule = {'action': 'Allow',
+                'source': {'selector': 'group=="n1"'}}
+        prof_n2['spec']['ingress'].append(rule)
+        rule = {'action': 'Allow',
+                'source': {'selector': 'group=="n2"'}}
+        prof_n1['spec']['ingress'].append(rule)
+        self._apply_new_profile(self.new_profiles, self.host1)
+        self.assert_connectivity(retries=2,
```

Comment:
Is it worth asserting connectivity doesn't work prior to applying the new netset rules?

Reply:
Yes; I thought that was covered in the setUp but maybe not;  I'll add something.